products-documentation-ingestion/requirements.txt

```diff
@@ -1,68 +1,13 @@
-einops==0.8.1; python_version >= '3.8'
-frozenlist==1.5.0; python_version >= '3.8'
-greenlet==3.1.1; python_version >= '3.7'
-grpcio==1.67.1; python_version >= '3.8'
-h11==0.14.0; python_version >= '3.7'
-html2text==2024.2.26; python_version >= '3.8'
-httpcore==1.0.7; python_version >= '3.8'
-httpx==0.28.1; python_version >= '3.8'
-httpx-sse==0.4.0; python_version >= '3.8'
-idna==3.10; python_version >= '3.6'
-jiter==0.8.2; python_version >= '3.8'
-jsonpatch==1.33; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4, 3.5, 3.6'
-jsonpointer==3.0.0; python_version >= '3.7'
-langchain==0.3.19; python_version >= '3.9' and python_version < '4.0'
-langchain-community==0.3.17; python_version >= '3.9' and python_version < '4.0'
-langchain-core==0.3.36; python_version >= '3.9' and python_version < '4.0'
-langchain-milvus==0.1.7; python_version >= '3.9' and python_version < '4.0'
-langchain-text-splitters==0.3.6; python_version >= '3.9' and python_version < '4.0'
-langsmith==0.3.8; python_version >= '3.9' and python_version < '4.0'
-lxml==5.3.1; python_version >= '3.6'
-marshmallow==3.26.1; python_version >= '3.9'
-milvus-lite==2.4.11; python_version >= '3.7'
-multidict==6.1.0; python_version >= '3.8'
-mypy-extensions==1.0.0; python_version >= '3.5'
-numpy==1.26.4; python_version >= '3.9'
-openai==1.63.2; python_version >= '3.8'
-orjson==3.10.15; python_version >= '3.8'
-packaging==24.2; python_version >= '3.8'
-pandas==2.2.3; python_version >= '3.9'
-propcache==0.2.1; python_version >= '3.9'
-protobuf==5.29.3; python_version >= '3.8'
-pydantic==2.10.6; python_version >= '3.8'
-pydantic-core==2.27.2; python_version >= '3.8'
-pydantic-settings==2.7.1; python_version >= '3.8'
-pymilvus==2.4.10; python_version >= '3.8'
-pypdf==5.3.0; python_version >= '3.8'
-python-dateutil==2.9.0.post0; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2'
-python-dotenv==1.0.1; python_version >= '3.8'
-pytz==2025.1
-pyyaml==6.0.2; python_version >= '3.8'
-requests==2.32.3; python_version >= '3.8'
-requests-toolbelt==1.0.0; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
-setuptools==75.8.0; python_version >= '3.9'
-six==1.17.0; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2'
-sniffio==1.3.1; python_version >= '3.7'
-soupsieve==2.6; python_version >= '3.8'
-sqlalchemy==2.0.38; python_version >= '3.7'
-tenacity==9.0.0; python_version >= '3.8'
-tqdm==4.67.1; python_version >= '3.7'
-typing-extensions==4.12.2; python_version >= '3.8'
-typing-inspect==0.9.0
-tzdata==2025.1; python_version >= '2'
-ujson==5.10.0; python_version >= '3.8'
-urllib3==2.3.0; python_version >= '3.9'
-yarl==1.18.3; python_version >= '3.9'
-zstandard==0.23.0; python_version >= '3.8'
+beautifulsoup4==4.13.3
+einops==0.8.1
+html2text==2024.2.26
+langchain==0.3.19
+langchain-community==0.3.17
+langchain-milvus==0.1.7
+lxml==5.3.1
+openai==1.63.2
+pymilvus==2.4.10
+pypdf==5.3.0
+python-dotenv==1.0.1
+tqdm==4.67.1
+kfp==2.11.0
```

Comment:
Maybe the frozen requirements can be added as well, so that it will work even when sub-dependencies change (which has happened before.. 😩)